app/context/theme/index.tsx

```diff
@@ -33,22 +34,40 @@ export function getDefaultThemeByAppearance(): Theme {
 export const ThemeContext = createContext(getDefaultThemeByAppearance());
 const {Consumer, Provider} = ThemeContext;
 
+const themeCache = new Map<string, Theme>();
+let clearingThemeCache = false;
+const clearThemeCache = () => {
+    if (clearingThemeCache) {
+        return;
+    }
+    clearingThemeCache = true;
+    themeCache.clear();
+    setTimeout(() => {
+        clearingThemeCache = false;
+    }, 300);
```

Comment:
Curious about the need of this timeout.  What kind of race condition are we worried about?

Reply:
added